march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator.py

```diff
@@ -399,12 +397,49 @@ def export_gait(self):
         if should_mirror:
             mirror = self.gait.get_mirror(key_1, key_2)
             if mirror:
-                export_to_file(mirror, self.get_gait_directory())
+                self.export_to_file(mirror, self.get_gait_directory())
             else:
                 UserInterfaceController.notify("Could not mirror gait", "Check the logs for more information.")
                 return
 
-        export_to_file(self.gait, self.get_gait_directory())
+        self.export_to_file(self.gait, self.get_gait_directory())
+
+    @staticmethod
+    def export_to_file(gait, gait_directory):
+        if gait_directory is None or gait_directory == "":
+            return
+
+        # Name and version will be empty as it's stored in the filename.
+        subgait_msg = gait.to_subgait_msg()
+
+        output_file_directory = os.path.join(gait_directory,
+                                             gait.gait_name.replace(" ", "_"),
+                                             gait.subgait_name.replace(" ", "_"))
+        output_file_path = os.path.join(output_file_directory,
+                                        gait.version.replace(" ", "_") + ".subgait")
+
+        file_exists = os.path.isfile(output_file_path)
+        if file_exists:
+            overwrite_file = QMessageBox.question(None, 'File already exists',
+                                                  "Do you want to overwrite " + str(output_file_path) + "?",
+                                                  QMessageBox.Yes | QMessageBox.No)
+            if overwrite_file == QMessageBox.No:
+                return
+
+        rospy.loginfo("Writing gait to " + output_file_path)
+
+        try:
+            os.makedirs(output_file_directory)
+        except OSError:
+            if not os.path.isdir(output_file_directory):
+                raise
+
+        output_file = open(output_file_path, 'w')
+        output_file.write(str(subgait_msg))
```

Comment:
i would use with open, this prevents a corrupted file when an error occurs
```suggestion
with open(output_file_path, 'w') as file:
   file.write(str(subgait_msg))
```

Reply:
And don't forget to close it as well
```python
with open(output_file_path, 'w') as file:
    file.write(str(subgait_msg))
    file.close()
```

Reply:
Excellent suggestion @JorisWeeda. @Olavhaasie closing is not necssary, the file is automatically closed when with exists.